### PR TITLE
sysdump: Use --hubble-flows-timeout flag

### LINKED ISSUE #907: cilium sysdump: `--hubble-flows-timeout` flag is ignored
The `--hubble-flows-timeout` flag is ineffective:
```
$ ./cilium sysdump --hubble-flows-timeout=5s | ts "[%H:%M:%S]"
[...]
[19:29:37] 🔍 Collecting platform-specific data
[19:29:37] 🔍 Collecting Hubble flows from Cilium pods
[19:30:34] 🗳 Compiling sysdump
[19:30:35] ✅ The sysdump has been saved to /home/paul/cilium-cli/cilium-sysdump-20220608-192936.zip
```

It seems the flag is defined but never used:
```
$ git grep HubbleFlowsTimeout
internal/cli/cmd/sysdump.go:76: cmd.Flags().DurationVar(&sysdumpOptions.HubbleFlowsTimeout,
internal/cli/cmd/sysdump.go:77:         "hubble-flows-timeout", sysdump.DefaultHubbleFlowsTimeout,
sysdump/defaults.go:34: DefaultHubbleFlowsTimeout                = 5 * time.Second
sysdump/sysdump.go:54:  HubbleFlowsTimeout time.Duration
```

### PR DESCRIPTION
**Pull request by michi-covalent**
Timeout was hardcoded to be 5 seconds, and --hubble-flows-timeout was
not being used.

Fixes #907

Signed-off-by: Michi Mutsuzaki <michi@isovalent.com>

### CODE REVIEW

**Comment by michi-covalent** [comment]
tested with
```
./cilium sysdump --hubble-flows-timeout=20s --debug | ts "[%H:%M:%S]"
```

File: sysdump/sysdump.go [hunk 1]
```diff
@@ -1098,6 +1100,7 @@ func (c *Collector) submitHubbleFlowsTasks(_ context.Context, pods []*corev1.Pod
 			if err := c.WriteBytes(fmt.Sprintf(hubbleObserveFileName, p.Name), e.Bytes()); err != nil {
 				return fmt.Errorf("failed to write hubble stderr for %q in namespace %q: %w", p.Name, p.Namespace, err)
 			}
+			c.logDebug("Finished collecting hubble flows from %s/%s", p.Namespace, p.Name)
```
Comment (michi-covalent) [inline sysdump/sysdump.go:1103]: added a debug log here. initially i was looking at this debug log: https://github.com/cilium/cilium-cli/blob/475f31829e32d4554660044709bf21e872d95ac0/sysdump/sysdump.go#L943 but realized this is incorrect, it doesn't wait until the task completes. couldn't figure out how to fix this.

**Comment by michi-covalent** [comment]
debugging external workload failure. converting it to draft.

**Review by tklauser — APPROVED** [review]
`cilium sysdump` changes LGTM.

The external workloads failure looks unrelated and also seem to occur on `master`. I'd be fine merging this as is and tackle failing tests in a separate PR - whatever you prefer.

**Comment by michi-covalent** [comment]
thanks for the review @tklauser! i'll spend one more day on the external workload failure. once i merge the PR i tend to move on and forget about the CI issue 😹

**Comment by tklauser** [comment]
@michi-covalent I've also opened #910 to debug the CI issue, so we could also continue there. Whatever you prefer 😄

**Comment by michi-covalent** [comment]
i didn't do anything but now it's passing 😑

**Comment by michi-covalent** [comment]
ok let's use https://github.com/cilium/cilium-cli/pull/910 to track the external workload failure, not sure if we can reproduce it though 😢

**Pull request merged**